packages/template-retail-react-app/app/components/recommended-products/index.jsx

```diff
@@ -164,7 +188,9 @@ const RecommendedProducts = ({zone, recommender, products, title, shouldFetch, .
                     )
                 },
                 enableFavourite: true,
-                isFavourite: !!wishlist.findItemByProductId(product?.productId),
+                isFavourite: !!wishlist?.customerProductListItems?.find(
+                    (item) => item.productId === product?.productId
+                ),
                 onFavouriteToggle: (isFavourite) => {
                     const action = isFavourite ? addItemToWishlist : removeItemFromWishlist
```

Comment:
Hmm, I think the order here should be flipped.

```suggestion
                    const action = isFavourite ? removeItemFromWishlist : addItemToWishlist
```

Reply:
ah good catch!